File: src/lessons/welcome/array/array667/Array667Entity.java
package lessons.welcome.array.array667;

import static plm.core.ValueSerializer.*;

import plm.universe.bat.BatEntity;

public class Array667Entity extends BatEntity {

  public void run()
  {
    int count = getTestCount();
    for (int i = 0; i < count; i++) {
      Object[] param = (Object[])deserialize(getTest(i));
      setTestResult(i, serialize(array667(toPrimitive(toArrayOfType(param[0], Integer.class)))));
    }
  }

  /* BEGIN TEMPLATE */
  int array667(int[] nums)
  {
    /* BEGIN SOLUTION */
    int count = 0;
    // Note: iterate to length-1, so can use i+1 in the loop
    for (int i = 0; i < (nums.length - 1); i++) {
      if (nums[i] == 6) {
        if (nums[i + 1] == 6 || nums[i + 1] == 7) {
          count++;
        }
      }
    }
    return count;
    /* END SOLUTION */
  }
  /* END TEMPLATE */
}

File: src/lessons/welcome/array/array667/Array667Entity.scala
package lessons.welcome.array.array667

import plm.universe.bat.BatEntity
import plm.universe.bat.BatTest

class Array667Entity extends BatEntity {

    override def run() {
      import plm.core.ValueSerializer._

      val count = getTestCount()
      for (i <- 0 to count -1) {
        val param = deserialize(getTest(i)).asInstanceOf[Array[Object]]
      	setTestResult(i, serialize(array667(param(0).asInstanceOf[Array[Int]]) ))
	  }
	}

	/* BEGIN TEMPLATE */
	def array667(nums:Array[Int]): Int = {
		/* BEGIN SOLUTION */
	  var count=0
	  for (i <- 0 to nums.length-2)
	    if ((nums(i) == 6) && (nums(i+1)==6 || nums(i+1)==7))
	      count += 1
	  return count
		/* END SOLUTION */
	}
	/* END TEMPLATE */
}
